Infusionsoft/App.php

```diff
@@ -196,8 +230,16 @@ public function hasTokens(){
     }
 
     public function refreshTokens(){
-        $tokens = Infusionsoft_OAuth2::refreshToken($this->refreshToken);
-        $this->updateAndSaveTokens($tokens['access_token'], $tokens['refresh_token'], $tokens['expires_in']);
+        if (method_exists($this->tokenStorageProvider, 'refreshTokens')){
+            $tokens = $this->tokenStorageProvider->refreshTokens();
+            $this->accessToken = $tokens['accessToken'];

```

Comment:
This is also kind of weird.  I think it should return true / false, and then you call get tokens to get the updated tokens.  (Yes, you did follow the way it was written, it's wrong too IMO)